sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -930,6 +930,18 @@ class CodegenContext {
       // inline execution if only one block
       blocks.head
     } else {
+      if (Utils.isTesting) {
```

Comment:
Do we only do the assert in testing? Because passing global variables won't raise compile error, if we have any global variables passed in when not in testing, the codegen still work and may lead to wrong result.

Reply:
as you said, it _may_ lead, but likely it doesn't. Then I do think that the best option is to assert it only in testing, where this might help finding _potential_ bugs. In production it is an overkill to throw an exception for a situation which most likely is not a problem IMHO.

Reply:
Here is a [discussion](https://github.com/apache/spark/pull/19865#discussion_r156862608) about this testing.